File: 1047/1047.ex
input = IO.gets("") |> String.trim()
[h1, m1, h2, m2] = input |> String.split(" ") |> Enum.map(&String.to_integer/1)

if h2 <= h1 and m2 <= m1 do
    t = (((h2 + 24)* 3600) + (m2 * 60)) - ((h1 * 3600) + (m1 * 60))
    h = div(t, 3600)
    m = div((t - (h * 3600)), 60)

    IO.puts "O JOGO DUROU #{h} HORA(S) E #{m} MINUTO(S)"
else
    if m2 <= m1 do
        t = (((h2 - 1)* 3600) + ((m2 + 60) * 60)) - ((h1 * 3600) + (m1 * 60))
        h = div(t, 3600)
        m = div((t - (h * 3600)), 60)

        IO.puts "O JOGO DUROU #{h} HORA(S) E #{m} MINUTO(S)"
    else
        t = ((h2* 3600) + (m2 * 60)) - ((h1 * 3600) + (m1 * 60))
        h = div(t, 3600)
        m = div((t - (h * 3600)), 60)

        IO.puts "O JOGO DUROU #{h} HORA(S) E #{m} MINUTO(S)"
    end
end

File: 1047/1047.rb
h1, m1, h2, m2 = gets.split.map(&:to_i)

if h2 <= h1 && m2 <= m1
  h2 += 24
elsif m2 <= m1
  m2 += 60
  h2 -= 1
end

t = ((h2 * 3600) + (m2 * 60)) - ((h1 * 3600) + (m1 * 60))
h = t / 3600
m = (t - (h * 3600)) / 60

puts "O JOGO DUROU #{h.to_i} HORA(S) E #{m.to_i} MINUTO(S)"